payment_alipay/controllers/main.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+import logging
+import pprint
+import urllib2
+import werkzeug
+
+from openerp import http, SUPERUSER_ID
+from openerp.http import request
+from openerp.addons.payment_alipay.models import util
+
+_logger = logging.getLogger(__name__)
+
+
+class AlipayController(http.Controller):
+    _notify_url = '/payment/alipay/notify/'
+    _return_url = '/payment/alipay/return/'
+    _cancel_url = '/payment/alipay/cancel/'
+
+    def alipay_validate_data(self, **post):
+        res = False
+        cr, uid, context = request.cr, request.uid, request.context
+        _KEY = request.registry['payment.acquirer']._get_alipay_partner_key()
+        _, prestr = util.params_filter(post)
+        mysign = util.build_mysign(prestr, _KEY, 'MD5')
+        if mysign != post.get('sign'):
+            return 'false'
+
+        reference = post.get('out_trade_no')
+        notify_id = post.get('notify_id')
+        seller_id = post.get('seller_id')
+        tx = None
+        if reference:
+            tx_ids = request.registry['payment.transaction'].search(cr, uid, [('reference', '=', reference)], context=context)
```

Comment:
All your code must be PEP8 compliant (https://www.python.org/dev/peps/pep-0008/)